lib/ansible/module_utils/network/nxos/utils/telemetry/telemetry.py

```diff
@@ -0,0 +1,186 @@
+        for sensor in module_params['subscriptions']:
+            if sensor['id'] == resource_key:
+                mp['subscriptions'].append(sensor)
+
+    return mp
+
+
+def valiate_input(playvals, type, module):
+    """
+    Helper method to validate playbook values for destination groups
+    """
+    if type == 'destination_groups':
+        if not playvals.get('id'):
+            msg = "Invalid playbook value: {0}.".format(playvals)
+            msg = msg + " Parameter <id> under <destination_groups> is required"
+            module.fail_json(msg=msg)
+        if playvals.get('destination') and not isinstance(playvals['destination'], dict):
+            msg = "Invalid playbook value: {0}.".format(playvals)
+            msg = msg + " Parameter <destination> under <destination_groups> must be a dict"
+            module.fail_json(msg=msg)
+        if not playvals.get('destination') and len(playvals) > 1:
+            msg = "Invalid playbook value: {0}.".format(playvals)
+            msg = msg + " Playbook entry contains unrecongnized parameters."
+            msg = msg + " Make sure <destination> keys under <destination_groups> are specified as follows:"
+            msg = msg + " destination: {ip: <ip>, port: <port>, protocol: <prot>, encoding: <enc>}}"
+            module.fail_json(msg=msg)
+
+    if type == 'sensor_groups':
+        if not playvals.get('id'):
+            msg = "Invalid playbook value: {0}.".format(playvals)
+            msg = msg + " Parameter <id> under <sensor_groups> is required"
+            module.fail_json(msg=msg)
+        if playvals.get('path') and 'name' not in playvals['path'].keys():
+            msg = "Invalid playbook value: {0}.".format(playvals)
+            msg = msg + " Parameter <path> under <sensor_groups> requires <name> key"
+            module.fail_json(msg=msg)
+
+
+def get_instance_data(key, cr_key, cr, existing_key):
+    """
+    Helper method to get instance data used to populate list structure in config
+    fact dictionary
+    """
+    data = {}
+    if existing_key is None:
+        instance = None
+    else:
+        instance = cr._ref[cr_key]['existing'][existing_key]
+
+    if key == 'destination_groups':
+        m = re.search(r"destination-group (\d+)", cr._ref['_resource_key'])
+        instance_key = m.group(1)
+        data = {'id': instance_key, cr_key: instance}
+
+    if key == 'sensor_groups':
+        m = re.search(r"sensor-group (\d+)", cr._ref['_resource_key'])
+        instance_key = m.group(1)
+        data = {'id': instance_key, cr_key: instance}
+
+    if key == 'subscriptions':
+        m = re.search(r"subscription (\d+)", cr._ref['_resource_key'])
+        instance_key = m.group(1)
+        data = {'id': instance_key, cr_key: instance}
+
+    # Remove None values
+    data = dict((k, v) for k, v in data.items() if v is not None)
+    return data
+
+
+def cr_key_lookup(key, mo):
+    """
+    Helper method to get instance key value for mo
```

Comment:
What is `mo` again? module something I suppose...? :-)
Please clarify the doc header.

Reply:
`mo` stands for Manage Object.